diff --git a/node_modules/push-receiver/src/gcm/index.js b/node_modules/push-receiver/src/gcm/index.js
index 9260129..55f2bc4 100644
--- a/node_modules/push-receiver/src/gcm/index.js
+++ b/node_modules/push-receiver/src/gcm/index.js
@@ -79,11 +79,9 @@ async function postRegister({ androidId, securityToken, body, retry = 0 }) {
     form : body,
   });
   if (response.includes('Error')) {
-    console.warn(`Register request has failed with ${response}`);
-    if (retry >= 5) {
-      throw new Error('GCM register has failed');
+    if (retry >= 20) {
+      throw new Error(`GCM register has failed: ${response}`);
     }
-    console.warn(`Retry... ${retry + 1}`);
     await waitFor(1000);
     return postRegister({ androidId, securityToken, body, retry : retry + 1 });
   }
